napalm_panos/panos.py

```diff
@@ -31,11 +31,12 @@
 
 from napalm_base.base import NetworkDriver
 from netmiko import ConnectHandler
+from netmiko import __version__ as netmiko_version
 
 
 class PANOSDriver(NetworkDriver):
 
-    def __init__(self, hostname, username, password, timeout=60, optional_args=None):
+    def __init__(self, hostname, username, password='', timeout=60, optional_args=None):
```

Comment:
That's the issue, you could or you could not and we have to be consistent. We can't make things extremely complex because things vary from vendor to vendor. Otherwise, people is going to start falling into traps unknowingly.

We could argue that the signature shouldn't have had the `username` and `password` to begin with and I might agree but every driver has to have exactly the same method signatures and the same outputs. That's a hard requirement.

If we want to revisit how some signatures look I am fine with that but whatever we decide has to be implemented everywhere.